### ISSUE TITLE
Two messengers with same event name have event collision

### ISSUE THREAD

**Issue by jeetiss** [okotoki/figma-messenger#1]
You create nice library with great types support 👍🏻

I wanna migrate my plugin https://github.com/jeetiss/figlight from `storeon` to `figma-messenger`, but now I can't create 2 different messengers with some events, because events sends in both listeners when triggered in one.

### steps to reproduce

1. add code below to `/examples/figma-plugin/src/main/index.ts#L13` in repo
```ts
const messenger1 = createMainThreadMessenger<MainToIframe, IframeToMain>()
messenger1.on('heyFromIframe', handler = (msg: string) => {
  console.log('Second handler executes, but I don\'t trigger events for him.')
})
```
2. run plugin and click `send message`
3.  both handlers fire, but I expect only one 
<img width="733" alt="Снимок экрана 2019-12-28 в 21 13 19" src="https://user-images.githubusercontent.com/6726016/71547899-094cbf00-29b7-11ea-8df2-35f791fd1d14.png">

**Comment by opudalo** [comment]
Hi there! Happy holidays!

Lets discuss. 

I thought about it too, when originally wrote it. The thinking was:
– emitter usually is a single object, which both handles and emits events
– in our case, we have 2 messenger types on different sides (UI/main thread). So technically you don't have this 1-to-1 connection of (send->on) (if that makes sense). 
– thus its 2 separate instances, which operate in a same global space.
– thus all events are global (i.e. different messenger instances which subscribed to `eventName` will receive it).

Also, I haven't tested it locally, but it seems that your PR would work only if `id` is the same for both sides - UI and the main thread. Which sometimes is true, but it is unpredictable as `id` is just an internal counter and 0 on UI doesn't have to correspond to 0 on the main thread.

I've seen your other PR with named messengers. Maybe it would be a solution?
– messengers operate in global space by default. All events are global, and all handlers from different messengers fire. Current behavior.
– we introduce named messengers, where it would bring that 1-to-1 connection between UI and main thread. 

Will look something like this:

**Main thread side**.
```typescript
const messenger1 = createMainThreadMessenger()
const messenger2 = createMainThreadMessenger()

// named messenger
const messenger3 = createMainThreadMessenger('SPECIAL')
const messenger4 = createMainThreadMessenger('SPECIAL')

const handleGlobal = () => {
  console.log('received hey from iframe')
}
messenger1.on('heyFromIframe', handleGlobal)
messenger2.on('heyFromIframe', handleGlobal)

const handleSPECIAL = () => {
  console.log('received named hey from iframe')
}
messenger3.on('heyFromIframe', handleSPECIAL)
messenger4.on('heyFromIframe', handleSPECIAL)
```

**Iframe (UI) side.**
```typescript
createIframeMessenger().send('heyFromIframe') // messenger1 and messenger2 received event
createIframeMessenger('SPECIAL').send('heyFromIframe') // messenger3 and messenger4 received event 
```

What is your use case for a 1-to-1 relation?
And what do you think about the proposed named messengers API?

**Comment by jeetiss** [comment]
Hello 🙌🏻
Thank you. Happy Holidays to you too!

> would work only if id is the same for both sides - UI and the main thread. Which sometimes is true, but it is unpredictable

Yes, and because of that I added 'named' messengers

> 0 on UI doesn't have to correspond to 0 on the main thread

Yep, it will be work only if messengers will be created in some order in iframe and main tread

> What is your use case for a 1-to-1 relation?

I wanna create something like "shared state"  and use that abstraction for every UI element in plugin. Kinda `const [state, setState] = useSharedState('selection')`  on UI, and 
```
const selection = createMainThreadMessenger('selection')
selection.on('change', handler)
```
on main thread side.

> And what do you think about the proposed named messengers API?
It will be nice!

I have only one question:
Why did you add `id`'s for messengers, if they all operated in a same global space?

Thanks for you work 🙌🏻

**Comment by opudalo** [comment]
Ok, then I'll take your PR as a template, and will add `named` messengers feature. 
I used `id`s to keep track of handlers of a particular messenger. So that `messenger1.off('some')` will not remove listeners from `messenger2`.

**Comment by jeetiss** [comment]
> I used ids to keep track of handlers of a particular messenger. 

Ow, of course, thanks for clarification.
Can I help with something?

**Comment by opudalo** [comment]
Fixed in `figma-messenger@1.0.4`
